### PR TITLE
Refactor phoenix version

### PR DESCRIPTION
**Pull request by rhnonose**
- Move `web` folder inside `lib`
- Rename module prefixes to `CoursePlannerWeb`
- Update `coherence`'s version to work with phoenix 1.3
- Make adapting changes as necessary

### CODE REVIEW

**Comment by rhnonose** [comment]
@ghatighorias if you could help with the failing test, I have no idea why it's breaking.

**Comment by ghatighorias** [comment]
Jesuuuus 1million files are added :D :D :P

**Comment by rhnonose** [comment]
Sorry, forgot to ignore `node_modules` in the new location 😱

**Comment by rhnonose** [comment]
I hadn't merged with master yet @ghatighorias.
No conflicts hooray 😄

**Comment by ggpasqualino** [comment]
I think we should ignore the files in `/assets/priv/static/` those files are the previous `/priv/static`

**Comment by ggpasqualino** [comment]
I think `CoursePlannerWeb.Mailer.UserEmail` should be `CoursePlanner.Mailer.UserEmail` according to the folder structure

**Comment by ggpasqualino** [comment]
please check the file `compile` for the right command to build the assets

File: lib/course_planner_web/templates/coherence/layout/app.html.eex [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
   <body>
     <main role="main">
-      <%= render @view_module, @view_template, assigns %>
+      <%= render CoursePlannerWeb.Coherence.SessionView, @view_template, assigns %>
```
Comment (rhnonose) [inline lib/course_planner_web/templates/coherence/layout/app.html.eex]: An incompatibility with coherence, for some reason it's using the wrong view.

File: assets/gulpfile.js [hunk 2]
```diff
@@ -21,24 +21,24 @@ function reportChange ( event ) {
...
-  'web/static/css/**/*.scss*',
+  'css/**/*.css*',
+  'css/**/*.scss*',
 ];
 
-var jsEntryPath = 'web/static/js/app.js';
+var jsEntryPath = 'js/app.js';
 
 var jsWatchPaths = [
   'deps/phoenix/priv/static/phoenix.js',
```
Comment (rhnonose) [inline assets/gulpfile.js]: Indeed, missed it.

**Comment by ghatighorias** [comment]
i don't exactly get why we are -6% coverage @ggpasqualino @rhnonose

**Comment by rhnonose** [comment]
Coherence's new version generated some extra files, could be that.
We should probably ignore them, not sure.

**Comment by rhnonose** [comment]
Indeed, ignoring `schemas.ex` and `coherence_messages.ex` pulled the coverage to 86%.

**Pull request merged**